bin/backward-compatibility-test.js

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env node
+/* eslint-disable no-console */
+/* eslint-disable no-undef -- avoid errors on process */
+
+const os = require("os");
+const fs = require("fs");
+const path = require("path");
+const { execSync } = require("child_process");
+
+const INSTRUCTIONS =
+  "Usage: node bin/backward-compatibility-test.js <FE_GIT_REF> <BE_GIT_REF> <build|start|test>";
+
+const args = process.argv.slice(2);
+if (args.length !== 3) {
+  console.log(INSTRUCTIONS);
+  process.exit(1);
+}
+
+const FE_GIT_REF = args[0];
+const BE_GIT_REF = args[1];
+const COMMAND = args[2];
+
+const TMP_FOLDER = path.join(process.cwd(), ".tmp");
+const FE_FOLDER = path.join(TMP_FOLDER, "metabase-fe");
+const BE_FOLDER = path.join(TMP_FOLDER, "metabase-be");
+const JAR_PATH = path.join(BE_FOLDER, "target/uberjar/metabase.jar");
+
+const TEST_TO_RUN = [
+  "e2e/test/scenarios/dashboard/dashboard.cy.spec.js",
+  "e2e/test/scenarios/question/caching.cy.spec.js",
+];
+const COMMA_SEPARATED_TESTS_TO_RUN = TEST_TO_RUN.join(",");
+
+console.log(`Using frontend from ${FE_GIT_REF}`);
+console.log(`Using backend from ${BE_GIT_REF}`);
+console.log("---");
+console.log("To test locally run:");
+console.log(
+  `node bin/backward-compatibility-test.js ${FE_GIT_REF} ${BE_GIT_REF} <build|start|test>`,
+);
+console.log("---");
+console.log(`TMP_FOLDER: ${TMP_FOLDER}`);
+console.log(`FE_FOLDER: ${FE_FOLDER}`);
+console.log(`BE_FOLDER: ${BE_FOLDER}`);
+console.log(`JAR_PATH: ${JAR_PATH}`);
+console.log(`COMMA_SEPARATED_TESTS_TO_RUN: ${COMMA_SEPARATED_TESTS_TO_RUN}`);
+
+function printStep(message) {
+  console.log("---");
+  console.log(message);
+  console.log("---");
+}
+
+function executeCommand(command, cwd, envOverrides = {}) {
+  console.log(`Executing: ${command}${cwd ? " in " + cwd : ""}`);
+  try {
+    execSync(command, {
+      stdio: "inherit",
+      cwd: cwd,
+      env: { ...process.env, ...envOverrides },
+    });
+  } catch (error) {
+    console.error(`Failed to execute command: ${command}`);
+    process.exit(1);
+  }
+}
+
+function build() {
+  printStep("Cleaning up tmp folder...");
+  if (fs.existsSync(TMP_FOLDER)) {
```

Comment:
Since we're using `force: true` below, I believe we can simplify the code by removing this condition.